File: packages/core/src/agent/AgentEvent.ts
import { createId } from "../utils/createId";

export type AgentEventType =
  | "input_received"
  | "context_captured"
  | "memory_retrieved"
  | "privacy_checked"
  | "response_generated"
  | "action_suggested"
  | "feedback_recorded";

export interface AgentEvent {
  id: string;
  type: AgentEventType;
  sessionId: string;
  data?: unknown;
  timestamp: string;
}

export function createAgentEvent(
  type: AgentEventType,
  sessionId: string,
  data?: unknown
): AgentEvent {
  return {
    id: createId("event"),
    type,
    sessionId,
    data,
    timestamp: new Date().toISOString(),
  };
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/agent/AgentEvent.kt
package ai.mobileagentos.compose.agent

import ai.mobileagentos.compose.utils.IdGenerator

data class AgentEvent(
    val id: String,
    val type: AgentEventType,
    val sessionId: String,
    val timestamp: String,
    val data: Any? = null
) {
    companion object {
        fun create(
            type: AgentEventType,
            sessionId: String,
            data: Any? = null
        ): AgentEvent {
            return AgentEvent(
                id = IdGenerator.generate("event"),
                type = type,
                sessionId = sessionId,
                timestamp = System.currentTimeMillis().toString(),
                data = data
            )
        }
    }
}